compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -143,14 +144,25 @@ pub enum MirPhase {
     /// of the `mir_promoted` query), these promoted elements are available in the `promoted_mir`
     /// query.
     ConstPromotion = 2,
-    /// After this phase
-    /// * the only `AggregateKind`s allowed are `Array` and `Generator`,
```

Comment:
Ah... because this phase doesn't actually do this lowering.

I guess make `Deaggregator` do a phase change? After the deaggregator, only array and generator are allowed anymore